workflow/artifacts/git/git.go

```diff
@@ -19,15 +21,17 @@ import (
 	wfv1 "github.com/argoproj/argo/v3/pkg/apis/workflow/v1alpha1"
 )
 
-// GitArtifactDriver is the artifact driver for a git repo
-type GitArtifactDriver struct {
+// ArtifactDriver is the artifact driver for a git repo
+type ArtifactDriver struct {
```

Comment:
I know that the warning you get about type starting with package name is annoying, but I actually think it would better just to leave it now

Reply:
I have a very weak preference for following the warning and naming all `ArtifactDriver`. Definitely not a strong preference.

As it stands now some are called `ArtifactDriver` and some have their type in the time (e.g., `S3ArtifactDriver`). Whatever we choose we have to make sure it's consistent.